app/controllers/matchings_controller.rb

```diff
@@ -54,6 +58,111 @@ def destroy
     end
   end
 
+  def category
+    return "person_to_time"
+  end
+
+  def global_settings
+    return {"minutes": @project.duration}
+  end
+
+  def people
+    all_participants_emails = Participant.where(project_id: @project.id).pluck(:email)
+
+    people = []
+    for email in all_participants_emails
+      row = {"name": email, "match_degree": 1}
+      people.push(row)
+    end
+
+    return people
+  end
+
+  def timeslots
+    all_project_times = ProjectTime.where(project_id: @project.id).pluck(:date_time)
+
+    timeslots = []
+    for timeslot in all_project_times
+      timeslot_formatted = timeslot.utc.strftime('%Y-%m-%d %H:%M %p')
+      row = {"timestamp": timeslot_formatted, "tracks": 1}
+      timeslots.push(row)
+    end
+
+    return timeslots
+  end
+
+  def preferences
+    all_participants_ids = Participant.where(project: @project.id).pluck(:id)
+    all_project_time_ids = ProjectTime.where(project_id: @project.id).pluck(:id)
+
+    preferences = []
+    all_participants_ids.each do |participant_id|
+      all_project_time_ids.each do |project_time_id|
+        person = Participant.find_by(project_id: @project.id)['email']
```

Comment:
I think you meant to do `Participant.find_by(id: participant_id)`
I've started making changes, will push to your branch soon if you don't mind

Same thing with `ProjectTime.find_by` right below

Reply:
Also, in my opinion doing `participant.email` is nicer than `participant['email']` for auto-complete and debugging purposes (yay RubyMine), but not sure what the Rails convention is.